packages/expo-updates/android/src/main/java/expo/modules/updates/EnabledUpdatesController.kt

```diff
@@ -140,11 +144,24 @@ class EnabledUpdatesController(
     BuildData.ensureBuildDataIsConsistent(updatesConfiguration, databaseHolder.database)
     databaseHolder.releaseDatabase()
 
-    startupProcedure.run()
+    stateMachine.execute(startupProcedure)
   }
 
-  private fun relaunchReactApplication(callback: LauncherCallback) {
-    startupProcedure.relaunchReactApplication(true, callback)
+  private fun relaunchReactApplication(shouldRunReaper: Boolean, callback: LauncherCallback) {
+    val procedure = RelaunchProcedure(
+      context,
+      updatesConfiguration,
+      databaseHolder,
+      updatesDirectory,
+      fileDownloader,
+      selectionPolicy,
+      reactNativeHost,
+      { startupProcedure.launcher!! },
+      { currentLauncher -> startupProcedure.setLauncher(currentLauncher) },
```

Comment:
may want to store these somewhere outisde of the procedures, possibly move back to controller